framework/src/controller/bus.js

```diff
@@ -179,10 +179,6 @@ class Bus extends EventEmitter2 {
 	 * @throws {Error} If event name does not exist to bus.
 	 */
 	publish(eventName, eventValue) {
-		if (!this.getEvents().includes(eventName)) {
-			throw new Error(`Event ${eventName} is not registered to bus.`);
-		}
-
```

Comment:
Why to remove this check? Its important check for the architecture that modules must register events and can only publish those.

Reply:
This line has to be put it back with consideration of #3263 and #3154 
Main reason is Network module needs to support `network:chain:postBlock` without knowing where it's going to.
It's dynamically generated by `procedure`

Reply:
Network module can call action of any module or subscribe to their events, but it can't publish events of others.

Reply:
yes, but it's publishing events under their own namespace `network:*`

Reply:
https://github.com/LiskHQ/lisk-sdk/issues/3263#issuecomment-487477409

Reply:
A module should be allowed to publish custom events.

Upon thinking about it more, I don't think that there is any value in restricting what kinds of events a module can publish.

From my perspective, the only purpose of the feature (of explicitly declaring available events from the module via `get events() { }`) is documentation. My interpretation of the declaration is that it should mean ‘this module definitely supports those events’ but it should not mean `this module ONLY supports those events`.